src/layouts/AppLayout.tsx

```diff
@@ -1,8 +1,22 @@
-import { useMemo } from 'react';
+import { useMemo, useState } from 'react';
 import { Routes, Route } from 'react-router-dom';
 import { AppNavigation, Footer } from '@components/common';
+import { Dialog, DialogHeader, DialogBody, Typography, Button } from '@material-tailwind/react';
+import { CheckIcon } from '@heroicons/react/24/outline';
+
+import { useListenEvent } from '@hooks';
+import { useAuthMutation } from '@hooks';
 
 export const AppLayout: Component<{ menu: RouteMenu; child: RouteChild }> = ({ menu, child }) => {
+  const [openDialog, setOpenDialog] = useState<boolean>(false);
+  const handleOpenDialog = () => {
+    setOpenDialog(!openDialog);
+  };
+  const { logout } = useAuthMutation();
+  const handleLogout = () => {
```

Comment:
WTF? `logout.mutateAsync` returns a Promise. You must use `async-await`.

Reply:
![image](https://github.com/ngyngcphu/tick3d-fe/assets/109587630/a4af1d4f-0a25-4da6-9278-be012d76a4bc)